netlogo-gui/src/main/ide/CodeCompletionPopup.scala

```diff
@@ -87,48 +91,66 @@ case class CodeCompletionPopup(compiler: CompilerServices) {
   def displayPopup(): Unit = {
     for{eA <- editorArea} {
       isPopupEnabled = true
-      val token = compiler.getTokenAtPosition(eA.getText(), eA.getCaretPosition)
-      val position = if (token != null) token.start else eA.getCaretPosition
+      val tokenOption = getTokenTillPosition(eA.getText(), eA.getCaretPosition)
+      var position = 0
+      tokenOption match {
+        case Some(token) => position = token.start
+        case None => position = eA.getCaretPosition
+      }
```

Comment:
The above 5 lines can be simplified to:
`val position = tokenOption.map(_.start).getOrElse(eA.getCaretPosition)`